From c8f96ff9b85a7fa4281989a74ea36303d1a7db72 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Sat, 30 Sep 2023 18:20:44 +0200
Subject: [PATCH 500/954] feat(stm32mp21): update SRAM1 and RETRAM base address

On STM32MP21, SRAM1 is located in range 0x0E060000-0x0E06FFFF.
RETRAM range is 0x0E040000-0x0E05FFFF.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 9b98186d7b5694b07ff07532666d9df22d9487c8 feat(stm32mp21): update SRAM1 and RETRAM base address]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I0ac34a1efefa601207eb8e66d66f7b41c37a5441
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/356319
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index aa1a015132..8c9c5c70da 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -85,9 +85,17 @@
  ******************************************************************************/
 #define STM32MP_SYSRAM_BASE		U(0x0E000000)
 #define STM32MP_SYSRAM_SIZE		U(0x00040000)
+#if STM32MP21
+#define SRAM1_BASE			U(0x0E060000)
+#else /* STM32MP21 */
 #define SRAM1_BASE			U(0x0E040000)
+#endif /* STM32MP21 */
 #define SRAM1_SIZE_FOR_TFA		U(0x00010000)
+#if STM32MP21
+#define RETRAM_BASE			U(0x0E040000)
+#else /* STM32MP21 */
 #define RETRAM_BASE			U(0x0E080000)
+#endif /* STM32MP21 */
 #define RETRAM_SIZE			U(0x00020000)
 #define STM32MP_BACKUP_RAM_BASE		U(0x42000000)
 
-- 
2.53.0

